core/src/scene/sceneLoader.cpp

```diff
@@ -1049,6 +1024,69 @@ std::vector<StyleParam> SceneLoader::parseStyleParams(Node params, Scene& scene,
     return out;
 }
 
+StyleUniforms SceneLoader::parseStyleUniforms(const Node& value, Scene& scene) {
+    std::string type = "";
+    std::vector<UniformValue> uniformValues;
+    if (value.IsScalar()) { //float, int (bool), string (texture)
+        UniformValue uVal;
+        try {
+            uVal = value.as<float>();
+            type = "float";
+        } catch (const BadConversion& e) {
+            auto strVal = value.as<std::string>();
+            type = "int";
```

Comment:
These lines confused me too. An integer value in yaml will always coerce to a float in the preceding `try` block (even if explicitly tagged as an integer, but that's a whole other rant) so only string values (or Booleans written as strings) will enter the catch block. Variables declared as `int` in GLSL can't be used in conditionals, so the type here should probably be `bool` (unless I'm misunderstanding something).